src/plugins/ExternalLoaderPlugin.ts

```diff
@@ -0,0 +1,101 @@
+import Compiler = require('webpack/lib/Compiler');
+const CopyWebpackPlugin = require('copy-webpack-plugin');
+const HtmlWebpackIncludeAssetsPlugin = require('html-webpack-include-assets-plugin');
+
+export type ExternalDescriptor = {
+	/**
+	 * The path that will be used to load this module. This property is used to configure the build to defer to the
+	 * external loader.
+	 */
+	name?: string
+	/**
+	 * If this is a boolean, it indicates whether to inject this dependency into the application. If inject is set to
+	 * true, this dependency should be a script or stylesheet. If this dependency is a directory and contains one or
+	 * more stylesheets or scripts that  should be injected into the application inject can be set to a string or array
+	 * of strings that point to the resource(s) to be injected. Only scripts and stylehseets can be injected.
+	 */
+	inject?: boolean | string | string[];
+
+	/**
+	 * Optional property to indicate how this external should be loaded
+	 */
+	type?: string;
+
+	/**
+	 * This is used to specify the location, relative to the project root, from where the dependency should be copied.
+	 */
+	from?: string;
+
+	/**
+	 * This can be used to specify the location, relative to the externals folder, where the dependency should be copied.
+	 */
+	to?: string;
+};
+
+/**
+ * Describes an external dependency
+ */
+export type ExternalDep = string | ExternalDescriptor;
+
+export default class ExternalDojoLoaderPlugin {
+	private _externals: ExternalDep[];
+	private _outputPath: string;
+	private _pathPrefix: string;
+	private _hash: boolean;
+
+	constructor(options: {
```

Comment:
could make a nice type for options and clean the formatting a little in the codez